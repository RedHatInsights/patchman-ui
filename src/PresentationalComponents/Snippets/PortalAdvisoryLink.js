import ExternalLinkSquareAltIcon from '@patternfly/react-icons/dist/js/icons/external-link-square-alt-icon';
import propTypes from 'prop-types';
import React from 'react';
import { Flex, FlexItem } from '@patternfly/react-core';

const PortalAdvisoryLink = ({ advisory }) => {
    return (
        <a
            href={`https://access.redhat.com/errata/${advisory}`}
            target="__blank"
        >
            <Flex flex={{ default: 'inlineFlex' }}>
                <FlexItem spacer={{ default: 'spacerSm' }}>
                    <ExternalLinkSquareAltIcon />
                </FlexItem>
                <FlexItem spacer={{ default: 'spacerSm' }}>
                    View packages and errata at
                    access.redhat.com
                </FlexItem>
            </Flex>
        </a>

    );
};

PortalAdvisoryLink.propTypes = {
    advisory: propTypes.string
};

export default PortalAdvisoryLink;
