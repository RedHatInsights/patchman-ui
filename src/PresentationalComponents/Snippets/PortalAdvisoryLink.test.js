import PortalAdvisoryLink from './PortalAdvisoryLink';
import toJson from 'enzyme-to-json';

const wrapper = shallow(<PortalAdvisoryLink advisory = {'testAdvisory'} />);

describe('PortalAdvisoryLink', () => {
    it('Should match the snapshot', () => {
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
