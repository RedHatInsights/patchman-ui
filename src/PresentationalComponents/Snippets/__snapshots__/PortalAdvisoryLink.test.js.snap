// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PortalAdvisoryLink Should match the snapshot 1`] = `
<a
  href="https://access.redhat.com/errata/testAdvisory"
  target="__blank"
>
  <Flex
    flex={
      Object {
        "default": "inlineFlex",
      }
    }
  >
    <FlexItem
      spacer={
        Object {
          "default": "spacerSm",
        }
      }
    >
      <ExternalLinkSquareAltIcon
        color="currentColor"
        noVerticalAlign={false}
        size="sm"
      />
    </FlexItem>
    <FlexItem
      spacer={
        Object {
          "default": "spacerSm",
        }
      }
    >
      View packages and errata at access.redhat.com
    </FlexItem>
  </Flex>
</a>
`;
